### PR TITLE
Upgraded e2e/apimachinery/aggregator to 1.10 Sample API server

### LINKED ISSUE #63622: Flaky Test : [sig-api-machinery] Aggregator Should be able to support the 1.7 Sample API Server using the current Aggregator fails on "pull-kubernetes-e2e-gce" and "ci-kubernetes-e2e-gci-gce"
## Flaky Tests
* [pull-kubernetes-e2e-gce](https://k8s-gubernator.appspot.com/builds/kubernetes-jenkins/pr-logs/pull/59284/pull-kubernetes-e2e-gce/)
* [ci-kubernetes-e2e-gci-gce](https://k8s-gubernator.appspot.com/builds/kubernetes-jenkins/logs/ci-kubernetes-e2e-gci-gce/)

## Flakiest test 
[sig-api-machinery] Aggregator Should be able to support the 1.7 Sample API Server using the current Aggregator

## Failures in TestGrid
* [pull-kubernetes-e2e-gce](https://k8s-testgrid.appspot.com/presubmits-kubernetes-blocking#pull-kubernetes-e2e-gce&include-filter-by-regex=Aggregator%20Should%20be%20able%20to%20support%20th)
* [ci-kubernetes-e2e-gci-gce](https://k8s-testgrid.appspot.com/sig-api-machinery#gce)

## Triage link
More nuanced logs and details can be found at https://storage.googleapis.com/k8s-gubernator/triage/index.html?sig=api-machinery&job=ci-kubernetes-e2e-gci-gce&test=Aggregator%20Should%20be%20able%20to%20support%20the%201.7%20Sample%20AP

# Error
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/apimachinery/aggregator.go:76
gave up waiting for apiservice wardle to come up successfully
Expected error:
    <*errors.errorString | 0xc420131150>: {
        s: "timed out waiting for the condition",
    }
    timed out waiting for the condition
not to have occurred
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/apimachinery/aggregator.go:337

/priority failing-test
/priority important-soon
/kind flake
/sig-api-machinery
@kubernetes/sig-api-machinery-bugs 
/assign @liggitt 

@liggitt can you please help triage this issue?

### PR DESCRIPTION
**Pull request by yliaog**
**What this PR does / why we need it**:
Upgraded the e2e test "Should be able to support the 1.10 Sample API Server using the current Aggregator" from 1.7

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #63622

**Special notes for your reviewer**:

**Release note**:

```release-note

```

### CODE REVIEW

**Comment by fedebongio** [comment]
@dims yes, that's my fault - I've generated and uploaded the image while I was working on upgrading the test, the yliao took over, but the image was uploaded. I'll fix it soon.

**Comment by fedebongio** [comment]
That said, we also discussed having a test for 1.9, another one for 1.10, and so on for at least the -2 versions we support by policy. So in your opinion, which version should be in the master docker file?

**Comment by dims** [comment]
@fedebongio i am good whatever you all come up with :) (logically the image name should probably have version number of k/ and the docker file should probably start from 1.0 and get bumped when the image gets updated.)

**Comment by ixdy** [comment]
1.9 is about to become unsupported once 1.12 is released. should we be targeting something newer?

**Comment by fedebongio** [comment]
I think the ideal is to have a test for the 3 supported versions that we support for policy. And yes, this is taking so long, that by the time we merge this one, will be deprecated (will always be better than 1.7...).

File: test/e2e/apimachinery/aggregator.go [hunk 1]
```diff
@@ -47,7 +47,7 @@ import (
 	. "github.com/onsi/ginkgo"
 )
 
-var serverAggregatorVersion = utilversion.MustParseSemantic("v1.7.0")
+var serverAggregatorVersion = utilversion.MustParseSemantic("v1.9.0")
```
[the remark this answers is not in the document]
Comment (yliaog) [reply test/e2e/apimachinery/aggregator.go]: yes, changed to 1.10

**Comment by spiffxp** [comment]
@fedebongio Does this depend on or is this obviated by the changes that were going to land via https://github.com/kubernetes/kubernetes/pull/69233

File: test/e2e/apimachinery/aggregator.go [hunk 2]
```diff
@@ -107,7 +107,7 @@ func TestSampleAPIServer(f *framework.Framework, image string) {
 
 	namespace := f.Namespace.Name
 	context := setupServerCert(namespace, "sample-api")
-	if framework.ProviderIs("gke") {
+	if framework.ProviderIs("gke") || framework.ProviderIs("gce") {
```
Comment (spiffxp) [inline test/e2e/apimachinery/aggregator.go]: what's going on here that is provider specific? ie: why couldn't this run on aws

**Comment by yliaog** [comment]
removed the rbac hack, used the newly built 1.10 image, the test passes now.

**Comment by yliaog** [comment]
Depend on #69239

**Comment by yliaog** [comment]
after some in-depth investigations, it turns out the 1.10 sample-apiserver would need the following rbac permissions (in addition to the system:auth-delegator role). I'm updating the PR to add these.

================================
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  name: sample-apiserver
rules:
- apiGroups:
  - admissionregistration.k8s.io
  resources:
  - '*'
  verbs:
  - list
- apiGroups:
  - ''
  resources:
  - namespaces
  verbs:
  - list

**Review by cheftako — APPROVED** [review]
/lgtm

**Review by janetkuo — APPROVED** [review]

**Pull request merged**